### PR TITLE
patmat: Fix outer test making vs annotated types

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes scala/community-build#1400

### CODE REVIEW

File: test/files/neg/outer-ref-checks.check [hunk 1]
```diff
@@ -16,9 +16,12 @@ outer-ref-checks.scala:43: warning: The outer reference in this type test cannot
 outer-ref-checks.scala:48: warning: The outer reference in this type test cannot be checked at run time.
     case _: Inner => // unchecked warning
           ^
+outer-ref-checks.scala:53: warning: The outer reference in this type test cannot be checked at run time.
+    case _: (Inner @unchecked) => // warning suppressed
+          ^
```
Comment (lrytz) [inline test/files/neg/outer-ref-checks.check]: `@unchecked` should suppress the warning here, no?

Comment (dwijnand) [reply test/files/neg/outer-ref-checks.check]: I can be convinced so, but I wasn't 100% sure.  Does `@unchecked` just mean "don't give any warning"?  I thought it was only for some subset of warnings.  I'll look to fix this.

Comment (lrytz) [reply test/files/neg/outer-ref-checks.check]: I don't know about a "spec" for `@unchecked`, and I think it became what it is today organically...

This commit message is relevant: https://github.com/scala/scala/commit/e7e1c77092. Would be good to extend the scaladocs of unchecked.

**Comment by dwijnand** [comment]
> (Running the "alt" version gives you "java.lang.NoSuchMethodException:
> Test$$anon$1.loaderLibraryOnly()" (bug 2) and the result isn't the one
> it should be... (bug 3)...)

Seth made me realised that this is because matching a structural type is a lie (thus the warning that is emitted without `@unchecked`).  In the real code (in zinc) its usage is accompanied with handling NoSuchMethodException's.  I'll bring that to the test, and handle the warning suppression too.

**Review by lrytz — APPROVED** [review]

**Pull request merged**